File: src/Block/InsertBlock.java
/**
 * This file copyright (c) 2015-2016, Jing Fan, Adalbert Gerald Soosai Raj, and Jignesh M. Patel
 *
 * See the file CREDITS.txt in the root directory for details.
 **/

package Block;

/**
 * @brief The SQL block for  INSERT INTO SELECT FROM clause.
 */
public class InsertBlock extends Block {
  /**
   * @brief Constructor.
   * @param stage The string indicates the stage of this code block.
   * @param indent The indent level.
   * @param attr The attribute list from the table.
   * @param targetTb The name of the new table.
   * @param fromTb The name of the from table.
   */
  public InsertBlock(String stage,
                     int indent,
                     String attr,
                     String targetTb,
                     String fromTb) {
    super(stage, indent);
    this.append("INSERT INTO " + targetTb);
    this.append("SELECT " + attr);
    this.append("FROM " + fromTb + ";");
    this.sql = this.sb.toString();
  }

}

File: python/src/Blocks/InsertBlock.py
from Block import Block
class InsertBlock(Block):
    '''
    generates sql for an insert block
    '''
    
    def __init__(self,stage,indent,attr,targetTb,fromTb,pred=""):
        '''
        constructor: generates insert into sql statements
        @param stage: stage of the block
        @type stage: string
        @param indent: indent level
        @type indent: int
        @param attr: attribute list to select from
        @type attr: list
        @param targetTb:table name of the target table
        @type targetTb: string
        @param fromTb: table name of the from table
        @type fromTb: string
        '''
        super(InsertBlock,self).__init__(stage,indent)
        self.append("INSERT INTO " + targetTb)
        self.append("(SELECT " + attr)
        self.append("FROM " + fromTb )
        if(len(pred) > 0):
            tail = "WHERE "+pred+" "
        tail = tail + ");"
        self.append(tail)
        self.sql = self.sb